src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskAssignByIdControllerTest.java

```diff
@@ -71,5 +71,29 @@ public void should_return_a_204_when_assigning_a_task_by_id() {
         assertions.taskVariableWasUpdated(task.get("taskId"), "taskState", "assigned");
     }
 
+    @Test
+    public void should_return_a_401_when_the_user_did_not_have_any_roles() {
+        Map<String, String> task = common.setupTaskAndRetrieveIds();
+
+        String assigneeId = getAssigneeId(authorizationHeadersProvider.getCaseworkerBAuthorizationOnly());
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            task.get("taskId"),
+            new AssigneeRequest(assigneeId),
+            APPLICATION_JSON_VALUE,
+            APPLICATION_JSON_VALUE,
+            authorizationHeadersProvider.getLawFirmAAuthorization()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.UNAUTHORIZED.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("timestamp", lessThanOrEqualTo(LocalDateTime.now()
+                                                     .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
+            .body("error", equalTo(HttpStatus.UNAUTHORIZED.getReasonPhrase()))
+            .body("status", equalTo(HttpStatus.UNAUTHORIZED.value()))
+            .body("message", equalTo("User did not have sufficient permissions to perform this action"));
+    }
+
 }
 
```

Comment:
Can we also include FT's to cover the 2 scenarios for 403? I.e.

When the user's role assignments are retrieved from the Role Assignment service
And none of the user's Role Assignments have the MANAGE permission

&

When the assignee's role assignments are retrieved from the Role Assignment service
And none of the assignee's Role Assignments have the OWN or EXECUTE permission

Reply:
done